lib/bolt/inventory/inventory.rb

```diff
@@ -125,16 +125,25 @@ def match_wildcard?(wildcard, target_name, ext_glob: false)
       end
 
       # If target is a group name, expand it to the members of that group.
-      # Else match against targets in inventory by name or alias.
+      # Else match against groups and targets in inventory by name or alias.
       # If a wildcard string, error if no matches are found.
       # Else fall back to [target] if no matches are found.
       def resolve_name(target, ext_glob: false)
         if (group = group_lookup[target])
           group.all_targets
         else
-          # Try to wildcard match targets in inventory
-          # Ignore case because hostnames are generally case-insensitive
-          targets = groups.all_targets.select { |targ| match_wildcard?(target, targ, ext_glob: ext_glob) }
+          targets = []
+
+          # Find groups that match the glob
+          group_lookup.each do |name, grp|
+            next unless match_wildcard?(target, name, ext_glob: ext_glob)
+            targets += grp.all_targets.to_a
```

Comment:
It looks like this behavior existed prior to this PR. But am I reading this correctly in that: `Group#all_targets` returns a `Set` and this method (`resolve_name`) will return a `Set` of targets if the `target` parameter exactly matches a group name, or an array of targets otherwise?

Reply:
Ya, that's correct. I'm not really sure why it was written like that, but since it's only called in one place and immediately mapped it probably doesn't matter too much. I can update it to always return either an `Array` or `Set` if we want, though.

Reply:
I assume that's unintentional but I understand not wanting to change a method that may have public users so I'm fine if you are merging as-is, since it honors the existing weirdness. Though if we want to keep it might b nice to add a comment regarding return values at the top of the method.

Reply:
Updated to just have it always return an `Array`. Easy enough fix. :) Thanks for pointing out the weirdness!